File: ToolBag/ParseTool/src/parsetool/workflow/models/Translation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package parsetool.workflow.models;

import java.util.ArrayList;
import java.util.List;
import parsetool.models.common.TokenPair;
import parsetool.workflow.models.statements.Translation_statement;

/**
 *
 * @author Administrator
 */
public class Translation extends TokenPair {
    private List<Translation_statement> Translation_statements 
            = new ArrayList<Translation_statement>();

    /**
     * @return the Translation_statements
     */
    public List<Translation_statement> getTranslation_statements() {
        return Translation_statements;
    }

    /**
     * @param Translation_statements the Translation_statements to set
     */
    public void setTranslation_statements(List<Translation_statement> Translation_statements) {
        this.Translation_statements = Translation_statements;
    }
}

File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Translation.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

using System.Collections.Generic;
using CodeHelper.Core.Parse.ParseResults;
using CodeHelper.Core.Parse.ParseResults.Workflows.Statements;
namespace CodeHelper.Core.Parse.ParseResults.Workflows
{


    /**
     *
     * @author Administrator
     */
    public class Translation : TokenPair
    {
        public List<Translation_statement> Translation_statements { get; set; }

        public Translation()
        {
        }

        internal TranslationInfo Parse()
        {
            var rslt = new TranslationInfo();
            foreach (var s in this.Translation_statements)
            {
                rslt.Statments.Add(s.Parse());
            }
            return rslt;
        }
    }
}